MAD.Procore.StudioHoursUpload/Jobs/StudioHourUploadLogProducer.cs

```diff
@@ -32,7 +31,12 @@ public async Task ProduceStudioHoursUploadLogs()
                 .Where(y => y.Region == this.procoreConfig.Name)
                 .ToListAsync();
 
-            var studioHours = await this.studioHourClient.GetStudioHours();
+            //Retrieve last processed studio log and pass processed date time to GetStudioHours method
+            var lastStudioLog = await this.studioHourDbContext.StudioHourUploadLog
+                .OrderByDescending(x => x.ProcessedDate)
+                .FirstOrDefaultAsync(x => x.Region == this.procoreConfig.Name && x.ProcessedDate.HasValue && string.IsNullOrWhiteSpace(x.Error));
```

Comment:
Remove the x.ProcessedDate.HasValue and string.IsNullOrWhiteSpace checks. If the log exists at all (processed / unprocessed /errored / not errored) then we do not want to generate it again